debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/core/launch/CoreBuildLaunchBarTracker.java

```diff
@@ -109,7 +110,23 @@ private void setActiveBuildConfig(ILaunchMode mode, ILaunchDescriptor descriptor
 		lastDescriptor = descriptor;
 		lastTarget = target;
 
-		// Pick build config based on toolchain for target
+		/*
+		 * Core build projects do not work with the concept of active build
+		 * configurations, like managed build projects. Instead they rely
+		 * on the launch mode set in the launchBar. A core build launch
+		 * configuration looks at the launchBar mode to pick the build
+		 * configuration. Core build projects still have an active build
+		 * configuration, but it is hidden for the user and ignored for
+		 * launching.
+		 *
+		 * Core build launch configurations have no options to set, so it is
+		 * likely that users will use non core build launch configurations to
+		 * launch a core build project binary. Non core build launch
+		 * configurations typically launch the active build configuration. The
+		 * active build configuration needs to match the launchBar mode.
+		 */
```

Comment:
* Core build launch configurations have no options to set, so it is
		 * possible that users will use non core build launch configurations to
		 * launch a core build project binary. Non core build launch
		 * configurations typically launch the active build configuration. The
		 * active build configuration needs to match the launchBar launch mode.
		 */

Reply:
same as above.